### ISSUE TITLE
Suggestion: Auto-leave town or nation chat upon fall of town/nation

### ISSUE THREAD

**Issue by Zifirl** [LlmDl/Towny#2593]
I wasn't sure if this should be listed as a bug but I'm assuming it's not one so I'll just leave it as a suggestion. Currently if a player has their town chat toggled and is offline while their town falls (not sure if this occurs if the player is online when the town falls), when they rejoin they won't be able to speak in general chat because their town chat is still toggled, and instead they'll get a message saying they can't speak in that channel. This is easily fixed by a player joining another town and then toggling town chat off, but it's often very confusing for players who come back after a long break and don't understand why they can't speak in chat. I guess my suggestion is for the plugin to automatically toggle off the town or nation chat if a player joins and is no longer in a town/nation, assuming that would be possible.

**Comment by LlmDl** [comment]
Are you using permission nodes to place players into their town chat by default when they log in?

**Comment by Zifirl** [comment]
No, but the players stay in the chat if they toggle it even when they log out, they'll still be in town chat when they log back in. And I was wrong, it doesn't send any message or give any error when you try to type in town/nation chat after your town/nation has been deleted. You just send the message and it doesn't appear in chat, no error message, just sends it but it doesn't get received.

**Comment by LlmDl** [comment]
Fixed in 0.91.4.8:

```
- Fix players who are in Town or Nation chat when their town is removed ending up in chat limbo.
```
